src/parser/mod.rs

```diff
@@ -1,20 +1,65 @@
+use btoi::{btoi, btou};
+use nom::{
+    branch::alt,
+    bytes::streaming::{tag, take_while_m_n},
+    character::{is_digit, streaming::crlf},
+    combinator::{map, map_res, value},
+    sequence::terminated,
+    IResult,
+};
 use std::fmt;
+
 mod ascii;
 pub use ascii::{parse_ascii_metadump_response, parse_ascii_response, parse_ascii_stats_response};
 
+mod meta;
+#[allow(unused_imports)]
+pub use meta::parse_meta_response;
+
 /// A value from memcached.
 #[derive(Clone, Debug, PartialEq)]
 pub struct Value {
     /// The key.
     pub key: Vec<u8>,
     /// CAS identifier.
     pub cas: Option<u64>,
+    /// NOTE: This is the client bitflags, not meta flags
+    /// which is an opaque number passed by the client
     /// Flags for this key.
-    ///
     /// Defaults to 0.
-    pub flags: u32,
+    pub flags: Option<u32>,
     /// Data for this key.
-    pub data: Vec<u8>,
+    pub data: Option<Vec<u8>>,
+    /// optional extra Meta Flags Response Data
+    /// Meta flags are not to be confused with client bitflags, which is an opaque
+    /// number passed by the client. Meta flags change how the command operates, but
+    /// they are not stored in cache.
+    pub meta_values: Option<MetaValue>,
+}
```

Comment:
Personally not a fan of the nested structure / making everything optional here, but it was easier to leave this as-is for this PR before refactoring so that these are decoupled.